controllers/hostingautoscalingpolicy/hostingautoscalingpolicy_controller.go

```diff
@@ -280,6 +289,11 @@ func (r *Reconciler) initializeContext(ctx *reconcileRequestContext) error {
 	return nil
 }
 
+// saveCurrentResourceIDsToStatus before updating the ResourceIDs
+func (r *Reconciler) saveCurrentResourceIDsToStatus(ctx *reconcileRequestContext) {
+	ctx.HostingAutoscalingPolicy.Status.ResourceIDList = ctx.ResourceIDList
```

Comment:
dont you need r.Status().Update here ?

Reply:
@RedbackThomson in other controllers I saw reconciler object is being updated whenever ctx object is being changed

Reply:
`ctx` is just a helper for us. I think it doesn't necessarily matter where you do it, it's up to us. I'm fine with either.

Reply:
I think we are relying on `updateStatusAndReturn` or `updateStatusAndError` otherwise.

Reply:
which is not being called here, let me see if this helps.

Reply:
I tried updating the method to use `r.Status().Update` similar to the updateStatus methods but unit tests still fail. Keeping as is for now since I have tested several cases with this implementation and am sure it works - 
```
// saveCurrentResourceIDsToStatus before updating the ResourceIDs
func (r *Reconciler) saveCurrentResourceIDsToStatus(ctx reconcileRequestContext) error {
	jobStatus := &ctx.HostingAutoscalingPolicy.Status
	jobStatus.ResourceIDList = ctx.ResourceIDList

	if err := r.Status().Update(ctx, ctx.HostingAutoscalingPolicy); err != nil {
		err = errors.Wrap(err, "Unable to update status")
		ctx.Log.Info("Error while updating status.", "err", err)
		return err
	}

	return nil
}
```